src/libraries/System.Net.WebSockets/src/System/Net/WebSockets/ManagedWebSocket.cs

```diff
@@ -151,12 +150,24 @@ internal sealed partial class ManagedWebSocket : WebSocket
         /// </summary>
         private object ReceiveAsyncLock => _utf8TextState; // some object, as we're simply lock'ing on it
 
-        /// <summary>Initializes the websocket.</summary>
-        /// <param name="stream">The connected Stream.</param>
-        /// <param name="isServer">true if this is the server-side of the connection; false if this is the client-side of the connection.</param>
-        /// <param name="subprotocol">The agreed upon subprotocol for the connection.</param>
-        /// <param name="keepAliveInterval">The interval to use for keep-alive pings.</param>
-        private ManagedWebSocket(Stream stream, bool isServer, string? subprotocol, TimeSpan keepAliveInterval)
+        private readonly WebSocketInflater? _inflater;
```

Comment:
We have been looking at the size of objects like these https://github.com/dotnet/runtime/pull/49270#discussion_r588968505

Something to consider when adding a bunch of fields like this.

For example, can the inflater hold the inflate buffer, inflate buffer position, and inflate buffer available?

cc @davidfowl

Reply:
Read my mind. I was wondering these could be moved into either the inflater or deflater or if those could both be merged into the same object and only created if compressed is turned on.

Reply:
Done :)